### PR TITLE
Add support for yaml file format

### PR DESCRIPTION
**Pull request by calebmer**

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -125,3 +126,24 @@ impl From<JSONError> for Error {
...
 }
+
+impl From<YAMLError> for Error {
+  fn from(error: YAMLError) -> Self {
+    match &error {
+      &YAMLError::Custom(ref message) => {
+        Error {
+          code: ErrorCode::BadRequest,
+          message: message.clone(),
+          hint: Some("Fix your YAML syntax.".to_string())
```
Comment (svmnotn) [inline src/error.rs]: I get that that is the error, but... better words.

**Comment by svmnotn** [comment]
Other than that, :+1:

File: src/error.rs (hunk 1 above)
Comment (calebmer) [inline src/error.rs]: Recommendation?

Comment (svmnotn) [inline src/error.rs]: Bad YMAL Synatx
Unparsable YMAL Synatx
There was an error with the YMAL Synatx

Something along those lines?

**Comment by calebmer** [comment]
Notice how I just left a reaction! This is awesome!

**Comment by svmnotn** [comment]
Sadly some of them are just squares on my machine... but, that is neat.

Other than the naming issue. this is good to merge.

File: src/error.rs (hunk 1 above)
Comment (calebmer) [inline src/error.rs]: Oh, the hint is a recommendation on how to fix the error. The message actually describes the error and comes from the internal library 😊

Comment (svmnotn) [inline src/error.rs]: Ok, how about. Something like

Your YMAL needs to be fixed
Check your YMAL syntax
Make sure your YMAL syntax is correct

**Pull request merged**